lib/fetchers/url.rb

```diff
@@ -24,6 +24,8 @@ def self.resolve(target, opts = {})
         resolve_from_string(target[:url], opts, target[:username], target[:password])
       elsif target.is_a?(String)
         resolve_from_string(target, opts)
+      elsif target.is_a?(URI)
+        resolve_from_string(target.to_s, opts)
```

Comment:
Hrm.  If I'm reading this right, the user passes a URI object, we render it to a String on line 28, call resolve_from_string, which parses the string into a URI object, on line 31?  That makes us rely on the round-tripping ability of URI, which seems risky.

Perhaps we should have a resolve_from_uri, which resolve_from_string calls?